keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepository.kt

```diff
@@ -156,25 +157,53 @@ class CombinedRepository(
   }
 
   /**
-   * Validates that resources have unique ids, throws an exception if invalid
+   * Run validation checks against delivery config to ensure:
+   *
+   * - resources have unique ids
+   * - artifacts have unique references
+   *
+   * Throws an exception if config fails any checks
    */
   private fun validate(config: DeliveryConfig) {
+
+    // helper function to get duplicates in a list
+    val duplicates: (List<String>) -> List<String> =
+      { ls -> ls.groupingBy { it }.eachCount().filter { it.value > 1 }.keys.toList() }
```

Comment:
I had trouble groking this at first glance. I think it's more readable as a `fun` with vertical spacing like:
```suggestion
    fun duplicates(resourceIds: List<String>) =
    	resourceIds.groupingBy { it }
        .eachCount()
        .filter { it.value > 1 }
        .keys
        .toList()
```

Reply:
Addressed in 1bea51c.